generators/app/templates/docs/coding-guides/build-specific-configurations.md

```diff
@@ -0,0 +1,164 @@
+examples:
+
+- test environment always builds without source maps, disables animations, uses a recaptcha test key, and disables
+  analytics
+- dev environment always builds with source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+- qa environment always builds with source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+- prod environment always builds without source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+
+### Limitations of Angular's fileReplacements
+
+But for certain deployment-specific configuration, things start to get really hairy, like in these examples:
+
+- QA build configuration needs to be built for local deployment, deployment to a server on the internet for QA
+  purposes, and also deployment to another server on the internet for staging purposes
+- Production build needs multiple different deployments of the same app to different servers
+
+These cases can cause problems when:
+
+- Each deployment needs a separate API URL
+- Each deployment needs a separate URL for building its own URLs to where it's deployed
+- Each deployment needs separate API keys, bucket names, etc
+
+You COULD start creating separate configurations for each deployment, each with its own fileReplacements, but that
+would be really messy.
+
+### Workarounds that don't work well
+
+One workaround would be to keep such configurations as globals in a separate deployment-specific script file. But
+that's pretty messy too. More importantly, there are limitations to where they can be used. For example, because
+of AOT, such configuration variables cannot be used in Angular's decorators, because they're not statically
+analyzable. So it would be better if we can keep everything in the same place.
+
+### ngx-rocket to the rescue
+
+The env task solves this problem really well, and avoids the need for separate environment.ts files for
+deployment-specific configuration. 
+
+To add a deployment-specific configuration:
+
+1. edit the existing environment.ts files for whichever environments you'd like to make that variable
+   deployment-specific for by having it come from the imported "env" object - pro tip: you can even make it fall
+   back to an environment-based default and still be statically analyzable!
+2. add that variable name to the npm script's env task
+
+Now, as long as you have that environment variable set in the shell running the build, the env task will save it into
+the .env file before building.
+
+If you really want, you can take things even further to the twelve-factor extreme, and you can even eliminate the
+need for fileReplacements entirely, and make all configuration come from environment variables.  Whether this will be
+the right approach for your project will be up to you.
+
+If you do decide to take that approach, your list of environment variables will probably get very long, so you can
```

Comment:
I think this is super abrupt and unclear.

I think before mentioning this, the bad news should be mentioned... what do you think of something like:

"
This makes separate deployments awesome and flexible, but unfortunately makes things a little bit of a hassle for your local development, test, etc. environments because you have the burden of providing all those keys, settings, etc. as environment variables.

To avoid having to do that, you'll probably want to ...
"

Thoughts?  Too obvious?

Reply:
Adding more context is a good idea, so readers have a better understanding of the why, so yes I think it would help to add this explanation 👍 (and obvious is good, as it may not be obvious to everyone)

Reply:
added :)